### PR TITLE
Set default headers to (created).

### PR DESCRIPTION
**Pull request by aljones15**
- [x] make (created) default headers
- [x] add positive tests for default headers
- [x] add negative tests for default headers
- [x] add tests for expires
- [x] look into why expires is set when options.headers contains it

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -247,23 +247,28 @@ api.createSignatureString = ({includeHeaders, requestOptions} = {}) => {
  * the request creating a signature & AuthzHeader.
  *
  * @param {Object} request - A request object.
- * @param {Object} options - Options for parsing such as clockSkew.
+ * @param {Object} [options={}] - Options for parsing such as clockSkew.
  *
  * @returns {Object} A parsed request with a signed header.
 */
 api.parseRequest = (request, options) => {
```
Comment (dlongley) [inline lib/index.js:254]:
```suggestion
api.parseRequest = (request, options = {headers: ['(created)']}) => {
```

File: lib/index.js [hunk 2]
```diff
@@ -247,23 +247,28 @@ api.createSignatureString = ({includeHeaders, requestOptions} = {}) => {
...
   assert.object(request, 'request');
+  // headers make up the major of coverted content used to create a signature
+  // so headers are required
   assert.object(request.headers, 'request.headers');
   if(options === undefined) {
     options = {};
   }
```
Comment (dlongley) [inline lib/index.js:261]:
```suggestion
```

File: lib/index.js [hunk 3]
```diff
@@ -247,23 +247,28 @@ api.createSignatureString = ({includeHeaders, requestOptions} = {}) => {
...
   if(options.headers === undefined) {
     options.headers = ['(created)'];
   }
+  // options are optional, but should always be an object
   assert.object(options, 'options');
+  // options.headers are also required, but is an array of covered content names
   assert.arrayOfString(options.headers, 'options.headers');
+  // clockSkew can't be infinite
   assert.optionalFinite(options.clockSkew, 'options.clockSkew');
```
Comment (dlongley) [inline lib/index.js]:
```suggestion
  // options are optional, but should always be an object
  assert.object(options, 'options');
  // options.headers are also required, but is an array of covered content names
  assert.arrayOfString(options.headers, 'options.headers');
  // clockSkew can't be infinite
  assert.optionalFinite(options.clockSkew, 'options.clockSkew');
```

File: lib/index.js [hunk 4]
```diff
@@ -247,23 +247,28 @@ api.createSignatureString = ({includeHeaders, requestOptions} = {}) => {
...
-
+  // This is the header that should contain the http signature
```
Comment (dlongley) [inline lib/index.js:271]:
```suggestion
  // this is the header that should contain the http signature
```

File: lib/index.js [hunk 5]
```diff
@@ -272,20 +277,24 @@ api.parseRequest = (request, options) => {
...
   options.clockSkew = options.clockSkew || 300;
+  // get the header with the http signature
   const authz = request.headers[authzHeaderName];
+  // parse the signature returning an object
   const parsed = parseSignatureHeader(authz);
 
-  // headers are optional and default to (created)
+  // the signature parameter headers is optional and defaults to (created)
   if(!parsed.params.headers) {
     // latest spec specifies this as the default headers
```
Comment (dlongley) [inline lib/index.js:287]:
```suggestion
    // latest spec (draft-ietf-httpbis-message-signatures-00) specifies this
    // as the default headers
```

**Review by dlongley — APPROVED** [review]
The additional comments are great, thanks! I left a few nits. Also, it seems that the defaults were already being set (I didn't see any changes around those, just new comments) -- so this PR just adds better docs via comments and better assertions, right?

File: lib/index.js (hunk 1 above)
Comment (aljones15) [reply lib/index.js:254]: I think `options = {}` would be better here as `options.clockSkew` and `options.headers` are added later.

File: lib/index.js (hunk 3 above)
Comment (aljones15) [reply lib/index.js]: let me know if this addresses this: https://github.com/digitalbazaar/http-signature-header/pull/24/commits/8dc1062bfb161d3ca74fdf8318140d8a0676dbfa

File: lib/index.js (hunk 2 above)
Comment (aljones15) [reply lib/index.js]: Addressed this here: https://github.com/digitalbazaar/http-signature-header/pull/24/commits/8dc1062bfb161d3ca74fdf8318140d8a0676dbfa

File: lib/index.js (hunk 1 above)
Comment (aljones15) [reply lib/index.js]: this is similar, but the default does not contain `headers`: https://github.com/digitalbazaar/http-signature-header/pull/24/commits/8dc1062bfb161d3ca74fdf8318140d8a0676dbfa

File: lib/index.js [hunk 6]
```diff
@@ -203,6 +203,17 @@ function lowerCaseObjectKeys(obj) {
   return newObject;
 }
 
+/**
+ * Takes in a list of covered content and a request
+ * and turns them into a canonicalized signature string.
+ *
+ * @param {Object} [options = {}] - Options to use.
```
Comment (dmitrizagidulin) [inline lib/index.js:210]:
```suggestion
 * @param {object} [options = {}] - Options to use.
```
All of these should be lowercase-o `{object}`, from what I understand.

**Review by dmitrizagidulin — APPROVED** [review]
👍 
aside from the jsdoc comment.

File: lib/index.js (hunk 6 above)
Comment (aljones15) [reply lib/index.js]: the eslint rule is javascript primitives are lower-case so `boolean`, `object`, `number`, `symbol`, `string` and then capitalize data structures composed from the primitives such as `Array` and `Function`. So yeah you are correct, but the linter is not throwing here... so I upgraded the linter and then linted all the files too. Thanks for catching this as the linter is a bit behind: https://github.com/digitalbazaar/http-signature-header/pull/24/commits/14216760f2d9028f177f3bfb3851ef9ee15e75e4

**Pull request merged**